examples/serialized.rs

```diff
@@ -39,19 +39,22 @@ fn main() -> anyhow::Result<()> {
     .invoke(Some(String::from("fib")))
     .func_args(vec![String::from("10")]) // This will generate 152 opcodes
     .build();
-  let mut wasm_ctx = WASMCtx::new_from_file(args)?;
 
-  let (proof, public_values, _) = ZKEngine::prove_wasm(&mut wasm_ctx)?;
+  let pp = ZKEngine::setup(&mut WASMCtx::new_from_file(&args)?)?;
+
+  // Use `BatchedZKEProof` for batched proving
+  let (proof, public_values, _) = ZKEngine::prove_wasm(&mut WASMCtx::new_from_file(&args)?, &pp)?;
 
   // Serialize the proof and public values
   let proof_str = serde_json::to_string(&proof)?;
   let public_values_str = serde_json::to_string(&public_values)?;
 
   // Deserialize the proof and public values
   let proof: ZKEProof<E1, BS1, S1, S2> = serde_json::from_str(&proof_str)?;
-  let public_values: PublicValues<E1, BS1, S1, S2> = serde_json::from_str(&public_values_str)?;
+  let public_values: PublicValues<E1> = serde_json::from_str(&public_values_str)?;
 
-  // Verify the proof
-  let result = proof.verify(public_values)?;
+  // Verify proof
+  let pp = ZKEngine::setup(&mut WASMCtx::new_from_file(&args)?)?;
+  let result = proof.verify(public_values, &pp)?;
```

Comment:
dq: how is `pp` different from `public_values`? I checked locally and bulk of the data is now in `pp` which is preferable but what does `public_values` contain at this point?

Reply:
Public values just contains the i/o of the proving system which is just 2 vecs of scalars values (one each for i/o )which are not very long/big.